giving-example/index.js

```diff
@@ -0,0 +1,312 @@
+const express = require("express");
+const path = require("path");
+const hbs = require("express-handlebars");
+const dotenv = require("dotenv");
+const morgan = require("morgan");
+const { uuid } = require("uuidv4");
+const { hmacValidator } = require('@adyen/api-library');
+const { Client, Config, CheckoutAPI } = require("@adyen/api-library");
+
+// init app
+const app = express();
+// setup request logging
+app.use(morgan("dev"));
+// Parse JSON bodies
+app.use(express.json());
+// Parse URL-encoded bodies
+app.use(express.urlencoded({ extended: true }));
+// Serve client from build folder
+app.use(express.static(path.join(__dirname, "/public")));
+
+// enables environment variables by
+// parsing the .env file and assigning it to process.env
+dotenv.config({
+  path: "./.env",
+});
+
+// Adyen Node.js API library boilerplate (configuration, etc.)
+const config = new Config();
+config.apiKey = process.env.ADYEN_API_KEY;
+const client = new Client({ config });
+client.setEnvironment("TEST");
+const checkout = new CheckoutAPI(client);
+
+app.engine(
+  "handlebars",
+  hbs.engine({
+    defaultLayout: "main",
+    layoutsDir: __dirname + "/views/layouts",
+    helpers: require("./util/helpers"),
+  })
+);
+
+app.set("view engine", "handlebars");
+
+/* ################# API ENDPOINTS ###################### */
+
+// Perform a donation
+app.post("/api/donations", async (req, res) => {
+
+  const donationToken = req.query.donationToken;
+  const pspReference = req.query.pspReference;
+
+  try {
+
+    const localhost = req.get('host');
+    const protocol = req.socket.encrypted? 'https' : 'http';    
+  
+    const response = await checkout.PaymentsApi.donations({
+      amount: { currency: req.body.currency, value: req.body.value },
+      reference: uuid(), 
+      paymentMethod: {
```

Comment:
@gcatanese I noticed that we need to pass-on the payment method here, is this something that a user can choose from? From the demo, it's not possible to switch payment methods: the backend needs to know what payment method was used, or we have to pass this data from the frontend.

Reply:
True, payment is hardcoded as the sample supports only Card